runtime/runtime/src/tests/apply.rs

```diff
@@ -2793,3 +2795,129 @@ fn test_deploy_and_call_local_receipts() {
         ActionErrorKind::FunctionCallError(FunctionCallError::MethodResolveError(_))
     );
 }
+
+#[test]
+fn test_transaction_batches_with_apply() {
+    let alice_key1 = InMemorySigner::test_signer(&alice_account());
+    let bob_signer = InMemorySigner::test_signer(&bob_account());
+    let alice_key2 = InMemorySigner::from_seed(alice_account(), KeyType::ED25519, "seed");
+
+    // Batch 1
+    let tx1 = SignedTransaction::send_money(
+        1,
+        alice_account(),
+        alice_account(),
+        &alice_key2, // invalid access key, tx should be dropped
+        100,
+        CryptoHash::default(),
+    );
+    // Batch 2
+    let tx2 = SignedTransaction::send_money(
+        1,
+        alice_account(),
+        alice_account(),
+        &alice_key1,
+        200,
+        CryptoHash::default(),
+    );
+    let tx3 = SignedTransaction::send_money(
+        2,
+        alice_account(),
+        bob_account(),
+        &alice_key1,
+        300,
+        CryptoHash::default(),
+    );
+    // Batch 3
+    let tx4 = SignedTransaction::send_money(
+        1,
+        bob_account(),
+        bob_account(),
+        &bob_signer,
+        400,
+        CryptoHash::default(),
+    );
+    let tx5 = SignedTransaction::send_money(
+        2,
+        bob_account(),
+        alice_account(),
+        &bob_signer,
+        500,
+        CryptoHash::default(),
+    );
+    let tx6 = SignedTransaction::send_money(
+        3,
+        bob_account(),
+        bob_account(),
+        &bob_signer,
+        600,
+        CryptoHash::default(),
+    );
+
+    let txs = vec![tx4.clone(), tx1, tx2.clone(), tx5.clone(), tx3.clone(), tx6.clone()];
+
+    // Construct transaction batches
+    let batches = TransactionBatches::new(&txs);
+    let batch_vec: Vec<TransactionBatch> = batches.par_batches().collect();
```

Comment:
```suggestion
    let batch_vec = batches.par_batches().collect:<Vec<_>>();
```

Less writing and it integrates better with IDEs.